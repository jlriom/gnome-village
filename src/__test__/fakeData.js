export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
