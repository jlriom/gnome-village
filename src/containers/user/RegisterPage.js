import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import {
	registerUser,
	loadHabitantsProfessions,
	loadHabitantsHairColors
} from "../../store/actions/userActions";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form";

const RegisterPage = ({
	professions,
	hairColors,
	error,
	show,
	handleClose,
	registerUser,
	loadHabitantsProfessions,
	loadHabitantsHairColors
}) => {
	const [values, setValues] = useState({
		name: "",
		thumbnail: "",
		age: 0,
		weight: 0,
		height: 0,
		hairColor: "",
		professions: []
	});

	useEffect(() => {
		loadHabitantsProfessions();
	}, [loadHabitantsProfessions]);

	useEffect(() => {
		loadHabitantsHairColors();
	}, [loadHabitantsHairColors]);

	const handleInputChange = e => {
		const { name, value } = e.target;
		setValues({ ...values, [name]: value });
	};

	const handleRegister = event => {
		const form = event.currentTarget;
		event.preventDefault();
		if (form.checkValidity() === false) {
			event.stopPropagation();
		} else {
			registerUser({
				name: values.name,
				thumbnail: values.thumbnail,
				age: values.age,
				weight: values.weight,
				height: values.height,
				hair_color: values.hairColor,
				professions: values.professions
			});
		}
	};

	return (
		<Modal show={show} onHide={handleClose} size='lg'>
			<Modal.Header closeButton>
				<Modal.Title>Register</Modal.Title>
			</Modal.Header>
			<Form onSubmit={handleRegister}>
				<Modal.Body>
					<div className='container'>
						<div className='row'>
							<div className='col-9'>
								<div className='row'>
									<div className='col-12'>
										<Form.Group controlId='name'>
											<Form.Label>Name</Form.Label>
											<Form.Control
												type='text'
												name='name'
												placeholder='Name'
												value={values.name}
												onChange={handleInputChange}
												required={true}
											/>
										</Form.Group>
									</div>
								</div>
								<div className='row'>
									<div className='col-12'>
										<Form.Group controlId='thumbnail'>
											<Form.Label>Photo</Form.Label>
											<Form.Control
												type='url'
												name='thumbnail'
												placeholder='Photo'
												value={values.thumbnail}
												onChange={handleInputChange}
												required={true}
											/>
										</Form.Group>
									</div>
								</div>
								<div className='row'>
									<div className='col-4'>
										<Form.Group controlId='age'>
											<Form.Label>Age</Form.Label>
											<Form.Control
												type='number'
												name='age'
												placeholder='Age'
												value={values.age}
												onChange={handleInputChange}
												required={true}
											/>
										</Form.Group>
									</div>
									<div className='col-4'>
										<Form.Group controlId='weight'>
											<Form.Label>Weight</Form.Label>
											<Form.Control
												type='number'
												name='weight'
												placeholder='Weight'
												value={values.weight}
												onChange={handleInputChange}
												required={true}
											/>
										</Form.Group>
									</div>
									<div className='col-4'>
										<Form.Group controlId='height'>
											<Form.Label>Height</Form.Label>
											<Form.Control
												type='number'
												name='height'
												placeholder='Height'
												value={values.height}
												onChange={handleInputChange}
												required={true}
											/>
										</Form.Group>
									</div>
								</div>
							</div>
							<div className='col-3'>
								<div className='row'>
									<div className='col-12'>
										<Form.Group controlId='exampleForm.ControlSelect2'>
											<Form.Label>Hair Color</Form.Label>
											<Form.Control as='select'>
												{hairColors.map(hairColors => (
													<option value={hairColors}>{hairColors}</option>
												))}
											</Form.Control>
										</Form.Group>
									</div>
								</div>
								<div className='row'>
									<div className='col-12'>
										<Form.Group controlId='exampleForm.ControlSelect2'>
											<Form.Label>Professions</Form.Label>
											<Form.Control as='select' multiple>
												{professions.map(profession => (
													<option value={profession}>{profession}</option>
												))}
											</Form.Control>
										</Form.Group>
									</div>
								</div>
							</div>
						</div>
					</div>
				</Modal.Body>

				<Modal.Footer>
					<Button variant='secondary' onClick={handleClose}>
						Cancel
					</Button>
					<Button variant='primary' type='submit'>
						Register
					</Button>
				</Modal.Footer>
			</Form>
		</Modal>
	);
};

const mapStateToProps = state => ({
	professions: state.user.professions,
	hairColors: state.user.hairColors,
	error: state.user.error
});

export default connect(
	mapStateToProps,
	{ registerUser, loadHabitantsProfessions, loadHabitantsHairColors }
)(RegisterPage);
