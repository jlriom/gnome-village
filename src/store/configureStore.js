import { createStore, applyMiddleware, compose } from "redux";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import thunk from "redux-thunk";
import logger from "redux-logger";

import rootReducer from "./reducers";

const middlewares = [thunk, logger, reduxImmutableStateInvariant()];

export default function configureStore(initialState) {
	const composeEnhancers =
		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(...middlewares))
	);
}
