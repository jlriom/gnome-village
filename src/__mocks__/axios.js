import * as fakeData from "./fakeData";

export default {
	get: jest.fn(() =>
		Promise.resolve({
			data: fakeData.habitants
		})
	),
	post: jest.fn(() => Promise.resolve({ data: fakeData.habitant })),
	put: jest.fn(() => Promise.resolve({ data: fakeData.habitant })),
	delete: jest.fn(() => Promise.resolve({ data: fakeData.habitant }))
};
