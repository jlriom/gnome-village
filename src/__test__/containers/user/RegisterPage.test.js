import React from "react";
import RegisterPage from "../../../containers/user/RegisterPage";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "../../../store/configureStore";
import { habitant } from "../../../__mocks__/fakeData";

const store = configureStore();

const props = {
	profile: habitant,
	professions: [],
	hairColors: [],
	show: false,
	handleClose: jest.fn(),
	loadProfile: jest.fn(),
	loadHabitantsProfessions: jest.fn(),
	loadHabitantsHairColors: jest.fn(),
	registerUser: jest.fn(),
	showError: jest.fn()
};

describe("<RegisterPage />", () => {
	it("renders ok", () => {
		const tree = renderer.create(
			<Provider store={store}>
				<RegisterPage {...props} />
			</Provider>
		);
		expect(tree).toMatchSnapshot();
	});
});
