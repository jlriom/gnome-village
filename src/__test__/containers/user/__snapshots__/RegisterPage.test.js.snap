// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterPage /> renders ok 1`] = `null`;
